sql/core/src/test/scala/org/apache/spark/sql/ArrowConvertersSuite.scala

```diff
@@ -124,8 +124,8 @@ class ArrowConvertersSuite extends SharedSQLContext {
 
   test("empty frame collect") {
     val arrowPayload = spark.emptyDataFrame.collectAsArrow()
-    assert(arrowPayload.nonEmpty)
-    arrowPayload.foreach(emptyBatch => assert(emptyBatch.getLength == 0))
+    assert(arrowPayload.isEmpty)
+    // TODO: test empty partitions
```

Comment:
Since ArrowPayload is an Iterator, I guess it makes sense that an empty DataFrame will convert to an empty Iterator.  However, if a partition is zero length, then it will convert that to a zero length ArrowRecordBatch.  Sound right?